tests/Http/Controllers/CelebTest.php

```diff
@@ -13,164 +14,334 @@ class CelebTest extends TestCase
     {
-        $celeb = \App\Models\Celeb::factory()->create();
+        $celeb = Celeb::factory()->create();
         $this->getJson("celebs/$celeb->id/edit")
-            ->assertStatus(403);
+            ->assertRedirect();
     }
 
     /** @test */
-    public function admin_can_access_celebs_update()
+    public function admin_can_update_a_celeb()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
-        $admin = \App\Models\User::factory()->create([
-            'is_admin' => true
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->putJson("celebs/$celeb->id", [
+                'name' => 'Jennifer Jaxon',
+                'date_of_birth' => '27/11/1982',
+                'photo' => 'https://generic/photo1 700x600'
+            ])->assertSessionHasNoErrors();
+
+        $this->assertDatabaseHas('celebs', [
+            'name' => 'Jennifer Jaxon',
+            'date_of_birth' => '27/11/1982',
+            'photo' => 'https://generic/photo1 700x600'
         ]);
+    }
+
+    /** @test */
+    public function validation_produces_errors_when_admin_updates_celeb_with_one_data_being_empty()
+    {
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
 
         $this->actingAs($admin)
-            ->putJson("celebs/$celeb->id")
-            ->assertOk();
+            ->put("celebs/$celeb->id", [
+                'name' => 'Olivia Plankton',
+                'date_of_birth' => '08/04/1994',
+                'photo' => null
+            ])->assertSessionHasErrors([
+                'photo' => 'The photo field is required.'
+            ]);
     }
 
     /** @test */
-    public function normal_user_cannot_access_celebs_update()
+    public function validation_produces_errors_when_admin_updates_celeb_with_all_data_being_empty()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
-        $user = \App\Models\User::factory()->create([
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->put("celebs/$celeb->id", [
+                'name' => null,
+                'date_of_birth' => null,
+                'photo' => null
+            ])->assertSessionHasErrors([
+                'name' => 'The name field is required.',
+                'date_of_birth' => 'The date of birth field is required.',
+                'photo' => 'The photo field is required.'
+            ]);
+    }
+
+    /** @test */
+    public function validation_error_when_admin_updates_celeb_with_date_of_birth_after_todays_date()
+    {
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->put("celebs/$celeb->id", [
+                'name' => 'James Islington',
+                'date_of_birth' => '08/03/2999',
```

Comment:
To best check this you should use Carbon, we usually avoid typed dates on tests. 

When using Carbon we also want to make sure to set up Carbon now(). To do that you have to add a setUp() method, like the example below:
```
 protected function setUp(): void
    {
        parent::setUp();

        Carbon::setTestNow(now());
    }
```

Then in your test you can use the Carbon::now()->addDay() or whatever function that you need you can check [Carbon docs ](https://carbon.nesbot.com/docs/)